CMakeLists.txt

```diff
@@ -53,9 +57,9 @@ if(HIOP_USE_MPI)
 endif(HIOP_USE_MPI)
 
 # The binary dir is already a global include directory
-configure_file(
-  "${CMAKE_SOURCE_DIR}/src/Interface/hiop_defs.hpp.in"
-  "${CMAKE_BINARY_DIR}/hiop_defs.hpp")
```

Comment:
I move these lines to line 236, since I may change the value of option `HIOP_SPARSE`. 
For example, user gives `HIOP_SPARSE=ON`, `HIOP_USE_COINHSL=OFF` and `HIOP_USE_STRUMPACK=OFF`, then there is no valid linear package for sparse linear algebra and I reset option `HIOP_SPARSE=OFF` internally.